sympy/solvers/solveset.py

```diff
@@ -564,7 +564,20 @@ def _solve_trig2(f, symbol, domain):
         denominators.append(Rational(c).q)
 
     x = Dummy('x')
-    mu = Rational(2)*ilcm(*denominators)/igcd(*numerators)
+
+    # ilcm() and igcd() require more than one argument
+    if len(denominators) == 1 and len(numerators) != 1:
+        mu = Rational(2)*denominators[0]/igcd(*numerators)
+
+    elif len(numerators) == 1 and len(denominators) != 1:
+        mu = Rational(2)*ilcm(*denominators)/numerators[0]
+
+    elif len(numerators) == 1 and len(denominators) == 1:
+        mu = Rational(2)*denominators[0]/numerators[0]
+
+    else:
+        mu = Rational(2)*ilcm(*denominators)/igcd(*numerators)
+
```

Comment:
Instead of the above, I would prefer the following:

```python
    try:
        mu = Rational(2)*ilcm(*denominators)/igcd(*numerators)
    except TypeError:
        if len(numerators) == 1 and len(denominators) == 1:
            mu = Rational(2)*denominators[0]/numerators[0]
        elif len(denominators) == 1:
            mu = Rational(2)*denominators[0]/igcd(*numerators)
        elif len(numerators) == 1:
            mu = Rational(2)*ilcm(*denominators)/numerators[0]
```
Its always better to handle such cases as exception, suppose if length of numerator and denominator is not 1 it has to check all the cases first.

Reply:
Since the two are the same length, and should not be empty, there is only one reason you will get the TypeError: when the length is 1. So we can just avoid it with one test:
```
if len(numerators) > 1:
    mu = Rational(2)*ilcm(*denominators)/igcd(*numerators)
else:
    assert len(numerators) == 1
    mu = Rational(2)*denominators[0]/numerators[0]
```

BUT...is there even a case where there is only 1 argument that is solved by trig2 that isn't solved by trig1?

Reply:
it's unlikely, can't think of a test case for now (Moreover, whenever I try to test an example `trigsimp` converts it into something else). So, It should only occur if trig1 throws an error due to any other reason and in that case, trig2 can return the solution.